plugins/module/create_kafka_acl_binding.py

```diff
@@ -0,0 +1,253 @@
+from rhoas_kafka_instance_sdk.api import acls_api
+from rhoas_kafka_instance_sdk.model.acl_binding import AclBinding
+from rhoas_kafka_instance_sdk.model.acl_resource_type import AclResourceType as art
+from rhoas_kafka_instance_sdk.model.acl_pattern_type import AclPatternType as apt
+from rhoas_kafka_instance_sdk.model.acl_operation import AclOperation as aot
+from rhoas_kafka_instance_sdk.model.acl_permission_type import AclPermissionType as apert
+
+
+import rhoas_kafka_instance_sdk
+from rhoas_kafka_instance_sdk.model.error import Error
+
+configuration = rhoas_kafka_instance_sdk.Configuration(
+    host = "https://api.openshift.com"
+)
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict(
+        kafka_id = dict(type='str', required = False),
+        principal = dict(type='str', required = True), 
+        kafka_admin_url = dict(type='str', required = False),
+        resource_type = dict(type='str', required = True),  
+        resource_name = dict(type='str', required = True), 
+        pattern_type = dict(type='str', required = True), 
+        operation_type = dict(type='str', required = True), 
+        permission_type = dict(type='str', required = True), 
+    )
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # changed is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+        original_message='',
+        message='',
+        kafka_admin_url='',
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifications
+    # module.check_mode = False
+    if module.check_mode:
+        module.exit_json(**result)
+
+    kafka_mgmt_config = rhoas_kafka_mgmt_sdk.Configuration(
+        host = "https://api.openshift.com",
+    )
+ 
+    token = auth.get_access_token()
+    
+    kafka_mgmt_config.access_token = token["access_token"]
+    
+    def get_kafka_mgmt_client():
+        with rhoas_kafka_mgmt_sdk.ApiClient(kafka_mgmt_config) as kafka_mgmt_api_client:
+            # Create an instance of the API class
+            kafka_mgmt_api_instance = default_api.DefaultApi(kafka_mgmt_api_client)
+            return kafka_mgmt_api_instance
+        
+    def retry_get_admin_url(kafka_mgmt_api_instance, retries = 10, backoff_in_seconds = 6):
+        attempt = 0
```

Comment:
Why attempts are neeed? Can we have  single failed response sent to user?

Reply:
How can we be certain of the length of time it takes to provision a kafka? Admin url is not available until the kafka is in a ready state right?

Reply:
It is not ACL binding responsibility to check for kafka status. It is.. create kafka role responsibility to do it.

Reply:
Good intentions - wrong place. Also we should check for status not url. Having url on success is implementation detail and it might change later. URL can be provided but kafka can be in failed state.

Reply:
Ah ok? So should we wait kafka to be in a ready state in the create_kafka module instead?

Reply:
![image](https://user-images.githubusercontent.com/981838/192061557-47a31216-6221-4273-8bb2-41312c14a8eb.png)